src/test/java/net/snowflake/client/jdbc/MultiStatementLatestIT.java

```
@@ -314,4 +322,72 @@ public void testAnonymousBlocksUse() throws SQLException {
       }
     }
   }
+
+  /**
+   * Tests multi-statements with different result formats (JSON and Arrow).
+   *
+   * @param queryResultFormat result format to test
+   * @throws Exception if failed
+   */
+  @ParameterizedTest
+  @ValueSource(strings = {"arrow", "json"})
+  void testMultiStatementResultFormat(String queryResultFormat) throws Exception {
+    // Setup a memory buffer for JUL logs
+    ByteArrayOutputStream logCapture = new ByteArrayOutputStream();
+    PrintStream logStream = new PrintStream(logCapture);
+
+    Logger sfLogger = Logger.getLogger("net.snowflake");
+    sfLogger.setLevel(Level.ALL);
+    sfLogger.setUseParentHandlers(false);
+
+    Handler consoleHandler =
+        new ConsoleHandler() {
+          @Override
+          public synchronized void publish(java.util.logging.LogRecord record) {
+            logStream.println(record.getLevel() + ": " + record.getMessage());
+          }
+        };
+    consoleHandler.setLevel(Level.ALL);
+    sfLogger.addHandler(consoleHandler);
+
+    try (Statement stmt = connection.createStatement()) {
+      try {
+        stmt.execute("alter session set jdbc_query_result_format = '" + queryResultFormat + "'");
+        stmt.execute("ALTER SESSION SET ENABLE_FIX_1758055_ADD_ARROW_SUPPORT_FOR_MULTI_STMTS=TRUE");
+      } catch (SQLException ignore) {
+        // test user may not have privilege to change params
+      }
+
+      stmt.unwrap(net.snowflake.client.jdbc.SnowflakeStatement.class)
+          .setParameter("MULTI_STATEMENT_COUNT", 4);
+
+      String multiStmtQuery =
+          "select TO_DOUBLE('0.123456789123456789');\n"
+              + "select 456;\n"
+              + "select 789;\n"
+              + "select '000';";
+
+      stmt.execute(multiStmtQuery);
+      ResultSet rs = stmt.getResultSet();
+
+      while (rs.next()) {
```

This isn't good. If query doesn't return any rows, your test silently passes. You should assert on this next.